bin/reth/src/commands/node/events.rs

```diff
@@ -155,19 +173,40 @@ impl<DB> NodeState<DB> {
             BeaconConsensusEngineEvent::ForkchoiceUpdated(state, status) => {
                 let ForkchoiceState { head_block_hash, safe_block_hash, finalized_block_hash } =
                     state;
+                if status != ForkchoiceStatus::Valid ||
+                    (self.safe_block_hash != Some(safe_block_hash) &&
+                        self.finalized_block_hash != Some(finalized_block_hash))
+                {
+                    info!(
+                        ?head_block_hash,
+                        ?safe_block_hash,
+                        ?finalized_block_hash,
+                        ?status,
+                        "Forkchoice updated"
+                    );
+                }
+                self.head_block_hash = Some(head_block_hash);
+                self.safe_block_hash = Some(safe_block_hash);
+                self.finalized_block_hash = Some(finalized_block_hash);
+            }
+            BeaconConsensusEngineEvent::CanonicalBlockAdded(block, elapsed) => {
                 info!(
-                    ?head_block_hash,
-                    ?safe_block_hash,
-                    ?finalized_block_hash,
-                    ?status,
-                    "Forkchoice updated"
+                    number=block.number,
+                    hash=?block.hash(),
+                    peers=self.num_connected_peers(),
```

Comment:
nit: should we display peers here? they're already in the `Status` message, and here it doesn't seem to be related to the block import

Reply:
Besu lists it, and that was something Afri added years ago.  I think parity listed it too in their lines.

As for status messages, if we are within a minute of sync they are suppressed because it's duplicate data (blocks come faster than the tick) - https://github.com/paradigmxyz/reth/pull/6095#discussion_r1481776119 - and peers was the only unique item. So it serves to keep the line count lower.

Reply:
ah, I see, right. Hmm, I just thought that it might be a bit confusing for users to see the number of peers alongside the block import -related data, as if peers are actually related to block import.